tests/api_client/test_api_client.py

```diff
@@ -0,0 +1,309 @@
+
+status_codes = requests.status_codes.codes
+
+
+@pytest.fixture
+def client():
+    return api_client.APIClient()
+
+
+SAMPLE_JOB_CREATE_RESPONSE = {
+    "id": 29583,
+    "status": "queued",
+    "result_url": "https://platform.xanadu.ai/jobs/29583/result",
+    "circuit_url": "https://platform.xanadu.ai/jobs/29583/circuit",
+    "created_at": "2019-05-24T15:55:43.872531Z",
+    "started_at": None,
+    "finished_at": None,
+    "running_time": None,
+}
+
+SAMPLE_JOB_RESPONSE = {
+    "id": 19856,
+    "status": "complete",
+    "result_url": "https://platform.xanadu.ai/jobs/19856/result",
+    "circuit_url": "https://platform.xanadu.ai/jobs/19856/circuit",
+    "created_at": "2019-05-24T15:55:43.872531Z",
+    "started_at": "2019-05-24T16:01:12.145636Z",
+    "finished_at": "2019-05-24T16:01:12.145645Z",
+    "running_time": "9µs"
+}
+
+
+class MockResponse:
+    status_code = None
+
+    def __init__(self, status_code):
+        self.status_code = status_code
+
+    def json(self):
+        return self.possible_responses[self.status_code]
+
+    @property
+    def text(self):
+        return json.dumps(self.json())
+
+    def raise_for_status(self):
+        raise requests.exceptions.HTTPError()
+
+
+class MockPOSTResponse(MockResponse):
+    possible_responses = {
+        201: SAMPLE_JOB_CREATE_RESPONSE,
+        400: {
+            "code": "parse-error",
+            "detail": (
+                "The blackbird script could not be parsed. "
+                "Please fix errors in the script and try again.")
+        },
+        401: {
+            "code": "unauthenticated",
+            "detail": "Requires authentication"
+        },
+        409: {
+            "code": "unsupported-circuit",
+            "detail": (
+                "This circuit is not compatible with the specified hardware.")
+        },
+        500: {
+            "code": "server-error",
+            "detail": (
+                "Unexpected server error. Please try your request again "
+                "later.")
+        },
+    }
+
+
+class MockGETResponse(MockResponse):
+    possible_responses = {
+        200: SAMPLE_JOB_RESPONSE,
+        401: {
+            "code": "unauthenticated",
+            "detail": "Requires authentication"
+        },
+        404: {
+            "code": "not-found",
+            "detail": "The requested resource could not be found or does not exist.",
+        },
+        500: {
+            "code": "server-error",
+            "detail": (
+                "Unexpected server error. Please try your request again "
+                "later.")
+        },
+    }
+
+    status_code = None
+
+    def __init__(self, status_code):
+        self.status_code = status_code
+
+    def json(self):
+        return self.possible_responses[self.status_code]
+
+    def raise_for_status(self):
+        raise requests.exceptions.HTTPError()
+
+
+@pytest.mark.api_client
+class TestAPIClient:
+    def test_init_default_client(self):
```

Comment:
Before merging, make sure to add a short 1-2 line docstring to each test function, just describing (1) what is being tested, and (2) the expected result. This shows up in pytest tracebacks so is useful for debugging.

Reply:
Done.